### DISCUSSION TITLE
SizeObserver does not return size

### DISCUSSION THREAD

**Discussion by karelvrabeckv** [Kitware/trame#500] [General] [answered]
Hello,

I installed my web application with the latest version of Trame. Suddenly, `SizeObserver` stopped to return the screen size:
`{'size': {}, 'pixelRatio': 1.25, 'dpi': 120, 'name': 'content_size'}`

```
@change("content_size")
def on_content_size_change(self, content_size, **kwargs):
    if content_size is None:
        return
    
    print(content_size)
    ...

...

with layout.content:
    with trame.SizeObserver("content_size"):
        with vuetify3.VContainer(fluid=True, classes="pa-0 fill-height"):
            view = vtk.VtkLocalView(
                ...
            )
            ...
```

I use `SizeObserver` for synchronizing the client with the server while picking points and cells.
Before, it returned width, height... Is this a bug in Trame?

Thank you very much!

**Comment by jourdain** [comment, marked as answer]
Wow, good catch. 

So it is fixed in the two latest `trame-client>=3.0.2` or `trame-component>=2.3.1` depending on where you are grabbing the SizeObserver from. It is has recently be added to the built-in trame client to make trame-component even more optional...

__For reference:__
The issue come from the fact that we were sending the size as a native JS object (DOMRect) which before was getting converted into a plain/json object over the network. But with the new network (wslink>=2), that conversion is not happening automatically (as we now can send inline binary structure) and require manual conversion. I'm expecting similar issue to happen with `trame>=3.6`, but hopefully we can find a middle ground to still support automatic JSON conversion for native JS object.
